src/stk/image/volume.h

```diff
@@ -135,11 +135,17 @@ namespace stk
         void set_origin(const float3& origin);
         void set_spacing(const float3& spacing);
         void set_direction(const Matrix3x3f& direction);
+        void set_direction(const std::initializer_list<float> direction);
 
         const float3& origin() const;
         const float3& spacing() const;
         const Matrix3x3f& direction() const;
 
+        const float3 index2point(const float3& index) const;
+        const float3 index2point(const int3& index) const;
+        const float3 point2index(const float3& point) const;
+        const float3 point2index(const int3& point) const;
```

Comment:
To make things clean, I thought it would be useful to have functions to convert between index space and image coordinate space. This also helps to cut quite some boilerplate code in the cost functions and similar. The lack of an implicit conversion between `float3` and `int3` is annoying though, for the moment I implemented two overloads, but I don't like it too much.

Reply:
Hmm, I see, I guess an implicit converter will cause a lot of problems in other places. Would be nice if we could stick to one type, but I don't think that's possible without making things really inconvenient...

Reply:
Yes, if we go for implicit conversions, that would backfire for sure. I was thinking it would be nice to have an explicit-only conversion operator though, so we can write something like `int3(p)` instead of `int3((int) p.x, (int) p.y, (int) p.z)` (or vice versa) every time we need a conversion...

Reply:
The problem is that if we add a constructor to int3 we break compliancy with the CUDA library. Could make a `make_int3` function though.

Reply:
That was my concern, paired with the fact that it would make the struct a non-POD type.